http/src/v0/refs.rs

```diff
@@ -138,33 +138,218 @@ async fn refs_paths<T: IpfsTypes>(
         walks.push(walk_path(&ipfs, path));
     }
 
-    let iplds = walks.try_collect().await?;
+    // strip out the path inside last document, we don't need it
+    let iplds = walks
+        .map_ok(|(cid, maybe_ipld, _)| (cid, maybe_ipld))
+        .try_collect()
+        .await?;
 
     Ok(iplds_refs(ipfs, iplds, max_depth, unique))
 }
 
+#[derive(Debug)]
+pub struct WalkError {
+    pub(crate) last_cid: Cid,
+    pub(crate) reason: WalkFailed,
+}
```

Comment:
this contextfull error type is ... quite painful. I think I'll need to redo it but unsure how.

Given the `reason` type which has `From` conversions from all of the nested types, I think there could be a:

```
trait WalkErrorContext {
  fn at(self, cid: Cid) -> WalkError;
}

impl<T: Into<WalkFailed>> WalkErrorContext for T {
  fn at(self, cid: Cid) -> WalkError {
    WalkError {
      last_cid: cid,
      reason: self.into(),
    }
  }
}
```

Or any other ideas? Obvious other solution is to have just the enum and make all choices contain the Cid. This other choice is at least a bit less painful but there's always the conversion..

Reply:
This would work but I don't think it makes the conversions any more readable. Any ideas?